openmetadata-ui/src/main/resources/ui/src/components/BotDetails/BotDetails.component.tsx

```diff
@@ -72,6 +77,11 @@ const BotDetails: FC<BotsDetailProp> = ({
   const [generateToken, setGenerateToken] = useState<boolean>(false);
   const [selectedExpiry, setSelectedExpiry] = useState('7');
 
+  const botPermissions = useMemo(
+    () => getPermissions(EntityType.BOT, permissions),
+    [permissions]
+  );
```

Comment:
Why can't we access botPermissions from Provider itself?

Reply:
@chirag-madlani , we have 35 resources should I create a method for all of them?

Reply:
That should reduce 4-5 lines in the component. so let's do it